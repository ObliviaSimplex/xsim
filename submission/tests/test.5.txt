=-=-=-=-=-=-=-=-=-=-=-=-=-=-
Test 5: stdout instructions
        cld, std, and out
=-=-=-=-=-=-=-=-=-=-=-=-=-=-

This test walks through a few output-related instructions. One surprising
bug that was found is that I had completely forgotten to put an entry for
cld into my jump table in xcpu.c! It just hadn't come up until now. This
was easily fixed.

Apart from that, the program behaved as expected, printing a jumble of
std information, interspersed with the messages

hi, how are you today?
fine

The output of ./xsim perfectly matched that of ./xsim_gold, and can be
found in test.5.out.

