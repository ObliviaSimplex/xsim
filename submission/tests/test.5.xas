# =-=-=-=-=-=-=-=-=-=-=-=-=-=-=
# Test 5: stdout instructions:
#         std, cld, and out
# =-=-=-=-=-=-=-=-=-=-=-=-=-=-=

.glob main

main:
        std
        push    r14
        mov     r15, r14

        cld
        loadi   0x68, r9
        loadi   0x69, r10
        out     r9
        out     r10
        loadi   readsomethinghere, r0     
        jr while
do:
        out      r2
        inc      r0
while:
        loadb    r0, r2 
        test     r2, r2
        br       do
        std
        loadi   writesomethinghere, r0
        loadi   0x0a, r6
        storb   r6, r0
        inc     r0
        loadi   0x66, r6
        storb   r6, r0
        inc     r0
        loadi   0x69, r6
        storb   r6, r0
        inc     r0
        loadi   0x6e, r6
        storb   r6, r0
        inc     r0
        loadi   0x65, r6
        storb   r6, r0
        inc     r0
        loadi   0x0a, r6
        storb   r6, r0
        inc     r0
        loadi   0x6, r10
        sub     r10, r0
        cld
        jr while2
do2:

        inc      r0
        out      r2
while2:
        loadb    r0, r2
        test     r2, r2
        br       do2        
        std
        xor     r0,  r0
        mov     r14, r15
        pop     r14
        ret

readsomethinghere:
.literal ", how are you today?"
writesomethinghere:
.words 0x30

