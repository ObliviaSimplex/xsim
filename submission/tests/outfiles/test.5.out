PC: 040c, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0408
PC: 040e, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406
PC: 0410, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0406 0406
hi, how are you today?PC: 0430, State: 0002: Registers:
 0496 0000 0000 0000 0000 0000 0000 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0434, State: 0002: Registers:
 0498 0000 0000 0000 0000 0000 0000 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0438, State: 0002: Registers:
 0498 0000 0000 0000 0000 0000 000a 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 043a, State: 0002: Registers:
 0498 0000 0000 0000 0000 0000 000a 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 043c, State: 0002: Registers:
 0499 0000 0000 0000 0000 0000 000a 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0440, State: 0002: Registers:
 0499 0000 0000 0000 0000 0000 0066 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0442, State: 0002: Registers:
 0499 0000 0000 0000 0000 0000 0066 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0444, State: 0002: Registers:
 049a 0000 0000 0000 0000 0000 0066 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0448, State: 0002: Registers:
 049a 0000 0000 0000 0000 0000 0069 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 044a, State: 0002: Registers:
 049a 0000 0000 0000 0000 0000 0069 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 044c, State: 0002: Registers:
 049b 0000 0000 0000 0000 0000 0069 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0450, State: 0002: Registers:
 049b 0000 0000 0000 0000 0000 006e 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0452, State: 0002: Registers:
 049b 0000 0000 0000 0000 0000 006e 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0454, State: 0002: Registers:
 049c 0000 0000 0000 0000 0000 006e 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0458, State: 0002: Registers:
 049c 0000 0000 0000 0000 0000 0065 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 045a, State: 0002: Registers:
 049c 0000 0000 0000 0000 0000 0065 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 045c, State: 0002: Registers:
 049d 0000 0000 0000 0000 0000 0065 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0460, State: 0002: Registers:
 049d 0000 0000 0000 0000 0000 000a 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0462, State: 0002: Registers:
 049d 0000 0000 0000 0000 0000 000a 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0464, State: 0002: Registers:
 049e 0000 0000 0000 0000 0000 000a 0000 0000 0068 0069 0000 0000 0000 0406 0406
PC: 0468, State: 0002: Registers:
 049e 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0406 0406
PC: 046a, State: 0002: Registers:
 0498 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0406 0406

fine
PC: 047a, State: 0002: Registers:
 049e 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0406 0406
PC: 047c, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0406 0406
PC: 047e, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0406 0406
PC: 0480, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0000 0408
PC: 0008, State: 0002: Registers:
 0000 0000 0000 0000 0000 0000 000a 0000 0000 0068 0006 0000 0000 0000 0000 040a
CPU has halted.
